usaspending_api/broker/management/commands/load_fpds_from_broker.py

```diff
@@ -0,0 +1,146 @@
+from django.core.management.base import BaseCommand
+import logging
+import re
+import psycopg2
+from datetime import datetime, timezone
+
+from usaspending_api.data_load.fpds_loader import run_fpds_load, destroy_orphans, load_chunk
+from usaspending_api.common.retrieve_file_from_uri import RetrieveFileFromUri
+from usaspending_api.common.helpers.date_helper import datetime_command_line_argument_type
+from usaspending_api.common.helpers.sql_helpers import get_broker_dsn_string
+from usaspending_api.common.helpers.etl_helpers import update_c_to_d_linkages
+from usaspending_api.etl.award_helpers import update_awards, update_contract_awards
+from usaspending_api.broker.helpers.last_load_date import get_last_load_date, update_last_load_date
+
+logger = logging.getLogger("console")
+
+BROKER_CONNECTION_STRING = get_broker_dsn_string()
+
+CHUNK_SIZE = 5000
```

Comment:
Per that article sent, memory could be a concern on larger batches given the width of these data tables, and how much is collected in the psycopg2 result set and then in the dictionaries before moving on to the next batch.

Reply:
While memory is a concern, we don't need to pursue the lowest memory consumption values as the article demonstrated. Batching is in-place to keep the jenkins worker memory low-enough, not necessarily low. The current loader fetches FPDS transactions from Broker in chuck sizes of 25,000, I don't expect any issues raising these batch sizes to 15,000.

If there is time to test performance and memory usage at different chunk sizes the numbers can be compared and an optimal chunk size can be determined.

Reply:
Is there a way to run those tests without just committing the code and trying it out? we can make this an option if we want which would let us try it much faster...

Reply:
The tests should run in Jenkins which implies committing code for jenkins to checkout.

Adding a cli param for chunk size is the first idea that comes to mind. Additional logging might be beneficial for this experimentation too

Reply:
The chunk size value doesn't need to change for the final code version. I'm curious if any testing was performed since the conversation started.

Reply:
I have tested with various sizes. Performance goes up as it increases (the peak is somewhere above 5K) with an asymptote somewhere around 100 records per second in local. 5K, however, finishes chunks in a clean minute with no memory scares, making jobs a little easier to watch and debug, without sacrificing that much performance. This could go up to 10 or 15 thousand, but I wouldn't predict more than 5-10% performance improvements.